code/src/main/kotlin/com/expediagroup/sdk/v2/core/extension/io/InputStreamExtension.kt

```diff
@@ -0,0 +1,21 @@
+package com.expediagroup.sdk.v2.core.extension.io
+
+import okio.Buffer
+import java.io.InputStream
+
+fun InputStream.toBuffer(contentLength: Int, exhaustStream: Boolean = true): Buffer {
+    val buffer = Buffer()
+
+    synchronized(this) {
```

Comment:
If you think about it, handling a response is usually done in a single thread. So do we want to do synchronization here?

Reply:
Fixed.